src/main/java/org/json/XML.java

```diff
@@ -445,6 +448,48 @@ private static boolean parse(XMLTokener x, JSONObject context, String name, XMLP
             }
         }
     }
+    /**
```

Comment:
What is the reasoning for trimming whitespace from CDATA? Is it a requirement for your project, or is there a more general reason?

Reply:
It is a more general reason. When whitespace is not trimmed (new behaviour), an issue occurs with inputs similar to the following: 
``` 
             "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"+
                        "<addresses>\n"+
                        "   <address>\n"+
                        "      <name> Sherlock Holmes </name>\n"+
                        "   </address>\n"+
                        "</addresses>";
```
Where whitespace between the closing and opening tags is incorrectly no longer trimmed and ends up in JSON with the key "content" (or whatever CDataTag is set to). The solution is to remove this value and key at this stage in the parsing if it is empty. 
At this stage the an empty value with tag CData is always going to be unintended unless the input itself has tags with the CData name, for this reason we have stated that inputs with the tags named the same as CDataTag are unsupported with the shouldTrimWhitespace set to false.

Reply:
@keatontaylor10 Setting a global flag changes the processing of the entire doc. Maybe it would be better to restrict the whitespace flag to specified tags, similar to how forceList works. Could you make that work for your project?